matchbox_socket/src/webrtc_socket/mod.rs

```diff
@@ -214,7 +222,11 @@ async fn message_loop<M: Messenger>(
                             .get_mut(&peer)
                             .expect("couldn't find data channel for peer")
                             .get_mut(channel_index).unwrap_or_else(|| panic!("couldn't find data channel with index {channel_index}"));
-                        data_channel.send(packet).unwrap();
+                        if let Err(e) = data_channel.send(packet) {
+                            // Peer is defunct, remove them
+                            error!("peer {peer} has become defunct: {e:?}");
+                            // TODO: do we need to remove the peer or do anything special here?
```

Comment:
Special attention to this - How can we remove the peer if we can no longer send them messages?

Reply:
Relevant? #287

Reply:
Is failure to send a message sufficient for determining that a peer has disconnected? What are the conditions under which this channel errors?

Reply:
I believe so. I think @johanhelsing has said sending fails if a future gets dropped or disconnected.

Reply:
This is properly handled now I think